### ISSUE TITLE
For how many epochs should we train the generator network?

### ISSUE THREAD

**Issue by cenalzw** [Rudrabha/Wav2Lip#126]
Thanks for your work. I am trying to reproduce your work and have been sucessful to train the lipsync discriminator. 
Now I am training the generator network.  I am just wondering how much L1 loss and Sync Loss should it be when we know the generator has converged. 

Looking forward to your answer, thank you!

**Comment by prajwalkr** [comment]
Both the L1 loss and the sync loss should go down to around 0.2-ish (0.21 for example).

**Comment by cenalzw** [comment]
@prajwalkr Thank you for the answer.

**Comment by eudoraxj** [comment]
Both the L1 loss and the sync loss should go down to around 0.2-ish (0.21 for example).

**Comment by eudoraxj** [comment]
Both the L1 loss and the sync loss should go down to around 0.2-ish (0.21 for example).
------
is the training loss? or the evaluation loss?

**Comment by prajwalkr** [comment]
> is the training loss? or the evaluation loss?

Both.

**Comment by wangzheliu** [comment]
@cenalzw Hi,how do you  train the lipsync discriminator?I train the lipsync discriminator but the loss  is fixed at 0.7, not decreasing.Can you tell me how do you deal with data.Thank you very much.

**Comment by prajwalkr** [comment]
The sync loss with the given hyper-parameters only starts converging after ~150K - 200K iterations. In our experiments it fluctuates around 0.69 till then.

**Comment by wangzheliu** [comment]
> 
> 
> The sync loss with the given hyper-parameters only starts converging after ~150K - 200K iterations. In our experiments it fluctuates around 0.69 till then.
I met the same situation. 0.69 is too high. Have you used this  0.69's model  test?

**Comment by wangzheliu** [comment]
> 
> 
> The sync loss with the given hyper-parameters only starts converging after ~150K - 200K iterations. In our experiments it fluctuates around 0.69 till then.
@prajwalkr I met the same situation. 0.69 is too high. Have you used this  0.69's model  test?

**Comment by prajwalkr** [comment]
After 200K iterations, it starts converging and converges to ~0.2 loss.

**Comment by wangzheliu** [comment]
> 
> 
> After 200K iterations, it starts converging and converges to ~0.2 loss.

when I train ,one global_step need 2 minute,200K  iterations need 280 days.I use 1080Ti.How long did you spend training lipsync discriminator

**Comment by prajwalkr** [comment]
The lip-sync discriminator trains at ~6-10it / sec on 1080Ti. In 2-3 days max, it converges.

**Comment by saharsh0612** [comment]
Training the lip sync descriminator with custom data, it is taking too long to complete one iterartion:
Loss: 0.6900650262832642: : 1it [00:25, 25.45s/it]
Loss: 0.6973084807395935: : 1it [00:26, 26.66s/it]
I am training on colab pro with Tesla P100 GPU.
Any suggestion would be appreciated on increasing the training speed.

**Comment by Utkarsh-shift** [comment]
L1: 0.025705222521679606, Sync: 0.21417364082270973, Percep: 0.8320787725883672 | Fake: 0.6512611059469708, Real: 0.6504838384107375: : 2477it [09:26,  4.65L1: 0.02570589264077559, Sync: 0.21415093795511395, Percep: 0.8320719455402642 | Fake: 0.6512701194477812, Real: 0.6504794993258177: : 2477it [09:26,  4.65iL1: 0.02570589264077559, Sync: 0.21415093795511395, Percep: 0.8320719455402642 | Fake: 0.6512701194477812, Real: 0.6504794993258177: : 2478it [09:26,  4.99iL1: 0.025705111478637573, Sync: 0.21416306389272574, Percep: 0.8320730458396536 | Fake: 0.6512595580176223, Real: 0.6505010755805538: : 2478it [09:27,  4.99L1: 0.025705111478637573, Sync: 0.21416306389272574, Percep: 0.8320730458396536 | Fake: 0.6512595580176223, Real: 0.6505010755805538: :    
is it converged ?
